frame/democracy/src/migrations.rs

```diff
@@ -0,0 +1,236 @@
+// This file is part of Substrate.
+
+// Copyright (C) 2022 Parity Technologies (UK) Ltd.
+// SPDX-License-Identifier: Apache-2.0
+
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// 	http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+//! Storage migrations for the preimage pallet.
+
+use super::*;
+use frame_support::{pallet_prelude::*, storage_alias, traits::OnRuntimeUpgrade, BoundedVec};
+use sp_core::H256;
+
+/// The log target.
+const TARGET: &'static str = "runtime::democracy::migration::v1";
+
+/// The original data layout of the democracy pallet without a specific version number.
+mod v0 {
```

Comment:
I'm not exactly sure why this couldn't go into `v1`, `v0` is kinda confusing as it's not a migration, but uses a migration naming scheme and sets data up to test v1 migrations.

Reply:
> […] as it's not a migration […]

What do you mean? It is a migration. It migrates the pallet from `v0` to `v1`. The name `v0` is arbitrary since the pallet did not have a version prior to `v1`.  
The storage alias are in `v0` since they access the storage state prio to `v1`.